packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php

```diff
@@ -42,11 +50,7 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext, Mess
             );
 
             return false;
-        }
-
-        $inventory = $this->inventoryRepository->getInventory($domain);
-        $group = $node instanceof DocReferenceNode ? 'std:doc' : 'std:label';
-        if (!$inventory->hasInventoryGroup($group)) {
+        } catch (InterlinkGroupNotFound) {
```

Comment:
I would recommend to not use exceptions for cases that are not really exceptions, performance wise exceptions have quit some impact on the execution time in php: https://php.watch/articles/php-exception-performance

It's not an exception that links cannot be found, it's a common flow of the code that a certain link cannot be resolved due misconfiguration.

Reply:
I learned that it is better to give a reason when something is not found (throw an expected exception) then just return null?

Reply:
I replaced them with the message system, please have another look